test/Microsoft.AspNet.Server.KestrelTests/SocketOutputTests.cs

```diff
@@ -199,6 +199,116 @@ public void WritesDontCompleteImmediatelyWhenTooManyBytesIncludingNonImmediateAr
         }
 
         [Fact]
+        public async Task OnlyWritesRequestingCancellationAreErroredOnCancellation()
```

Comment:
Hmm... you want to raise a CancellationToken issue to track? Will take about 5 mins to do a complete on error PR.